Gems/Atom/RPI/Code/Source/RPI.Reflect/Shader/ShaderOptionGroupLayout.cpp

```diff
@@ -109,8 +109,13 @@ namespace AZ
                     ->Attribute(AZ::Script::Attributes::Storage, AZ::Script::Attributes::StorageType::RuntimeOwn)
                     ->Method("GetName", &ShaderOptionDescriptor::GetName)
                     ->Method("GetDefaultValue", &ShaderOptionDescriptor::GetDefaultValue)
-                    ->Method("GetValueName", &ShaderOptionDescriptor::GetValueName)
+                    ->Method("GetValueName", static_cast<Name (ShaderOptionDescriptor::*)(ShaderOptionValue) const>(&ShaderOptionDescriptor::GetValueName))
                     ->Method("FindValue", &ShaderOptionDescriptor::FindValue)
+                    ->Method("GetMinValue", &ShaderOptionDescriptor::GetMinValue)
+                    ->Method("GetMaxValue", &ShaderOptionDescriptor::GetMaxValue)
+                    ->Method("GetValuesCount", &ShaderOptionDescriptor::GetValuesCount)
+                    ->Method("GetType", &ShaderOptionDescriptor::GetType)
+                    ->Method("GetValueNameWithInt", static_cast<Name (ShaderOptionDescriptor::*)(uint32_t) const>(&ShaderOptionDescriptor::GetValueName))
```

Comment:
Assuming the number represents an index...
```suggestion
                    ->Method("GetValueNameByIndex", static_cast<Name (ShaderOptionDescriptor::*)(uint32_t) const>(&ShaderOptionDescriptor::GetValueName))
```